x-pack/plugin/mapper-aggregate-metric/src/test/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapperTests.java

```diff
@@ -156,21 +148,42 @@ public void testParseEmptyValue() throws Exception {
         );
     }
 
-    /**
-     * Test parsing an aggregate_metric field that contains no values
-     * when ignore_malformed = true
-     */
-    public void testParseEmptyValueIgnoreMalformed() throws Exception {
-        DocumentMapper mapper = createDocumentMapper(
-            fieldMapping(
-                b -> b.field("type", CONTENT_TYPE)
-                    .field(METRICS_FIELD, new String[] { "min", "max", "value_count" })
-                    .field("ignore_malformed", true)
-                    .field(DEFAULT_METRIC, "max")
-            )
+    @Override
+    protected boolean supportsIgnoreMalformed() {
+        return true;
+    }
+
+    @Override
+    protected List<ExampleMalformedValue> exampleMalformedValues() {
+        return List.of(
+            // no metrics
+            exampleMalformedValue(b -> b.startObject().endObject()).matcher(
+                "Aggregate metric field [field] must contain all metrics [min, max, value_count]"
+            ),
+            // unmapped metric
+            exampleMalformedValue(
+                b -> b.startObject().field("min", -10.1).field("max", 50.0).field("value_count", 14).field("sum", 55).endObject()
+            ).matcher("Aggregate metric [sum] does not exist in the mapping of field [field]"),
+            // missing metric
+            exampleMalformedValue(b -> b.startObject().field("min", -10.1).field("max", 50.0).endObject()).matcher(
+                "Aggregate metric field [field] must contain all metrics [min, max, value_count]"
+            ),
+            // invalid metric value
+            exampleMalformedValue(b -> b.startObject().field("min", "10.0").field("max", 50.0).field("value_count", 14).endObject())
+                .matcher("Failed to parse object: expecting token of type [VALUE_NUMBER] but found [VALUE_STRING]"),
+            // negative value count
+            exampleMalformedValue(b -> b.startObject().field("min", 10.0).field("max", 50.0).field("value_count", -14).endObject()).matcher(
+                "Aggregate metric [value_count] of field [field] cannot be a negative number"
+            ),
+            // value count with decimal digits (whole numbers formatted as doubles are permitted, but non-whole numbers are not)
+            exampleMalformedValue(b -> b.startObject().field("min", 10.0).field("max", 50.0).field("value_count", 77.33).endObject())
+                .matcher("failed to parse field [field.value_count] of type [integer]")
         );
-        ParsedDocument doc = mapper.parse(source(b -> b.startObject("field").endObject()));
-        assertThat(doc.rootDoc().getField("field"), nullValue());
+    }
+
+    @Override
+    protected String[] ignoredFields() {
+        return new String[] { "field.value_count", "field.min", "field.max" };
```

Comment:
This is interesting - the `_ignored` field is really for user inspection, so I'm not sure if this mapper should be recording all its subfields in it as they are implementation details.

Reply:
Yeah! I think that's worth discussing somewhere. I can file an issue.

Reply:
#90572